### ISSUE TITLE
Issue with full screen mode in macos.

### ISSUE THREAD

**Issue by mayankdutta** [kovidgoyal/kitty#3892]
As soon as i switch to full screen mode in macos, Whatever i am typing doesn't show up on kitty terminal however kitty being working in the background like, if i type `exit` then kitty close.

**To Reproduce**
1. Install kitty via brew (`brew install kitty`)
2. open in full screen mode. 

**Video**

https://user-images.githubusercontent.com/33067129/127782011-dfd464b7-2576-4de4-b1cf-eb3a0e74be16.mov

_I was actually typing in full screen mode, but nothing showed up, and as i typed exit and pressed enter ... that worked._

**Environment details**
```
Press Ctrl+Shift+F6 (cmd+option+comma on macOS) in kitty, to copy debug output about kitty and its
configuration to the clipboard and paste it here. 

kitty 0.22.1 created by Kovid Goyal
Darwin Mayanks-MacBook-Air.local 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64
ProductName:	macOS ProductVersion:	11.4 BuildVersion:	20F71

Config options different from defaults:


On older versions of kitty, run kitty --debug-config instead
```

**Additional context**
1. Issue is same with or without config. 
2. If I build this from [source](https://sw.kovidgoyal.net/kitty/build/) then terminal runs fine just as expected. 
3. Problem occurred as i updated to the latest version.
4. Uninstalling then installing doesn't solve
5. kitty version 0.22.1

**Machine specs**
```
MacBook Air (13-inch, 2017)
1.8 GHz Dual-Core Intel Core i5
macos Big Sur 11.4
```

**Comment by Frederick888** [comment]
I suspect this was an issue fixed in 6bfb704f6f7dc3a35a1e118608ea9687eb8ca778? I was also having this issue so I compiled Kitty v0.22.1 as suggested however it still persisted. Then I built c797944 and the issue was gone.

**Comment by kovidgoyal** [comment]
dup of #3890
